### PR TITLE
More precise types

### PR DESCRIPTION
**Pull request by drets**

### CODE REVIEW

File: src/Network/MQTT/Packet.hs [hunk 1]
```diff
@@ -134,13 +134,13 @@ data PubcompPacket
...
     { subsribePacketIdentifier :: !PacketIdentifier,
-      subscribeTopicFilters :: [TopicFilter]
+      subscribeTopicFiltersQoS :: [(TopicFilter, QoS)]
     } deriving (Eq, Show)
 
 data SubackPacket
   = SubackPacket
     { subackPacketIdentifier :: !PacketIdentifier,
-      subackResponses :: [Maybe QoS]
+      subackResponses :: [QoS]
```
Comment (rasendubi) [inline src/Network/MQTT/Packet.hs]: Should be `Maybe QoS` as there is special value `Failure` (`0x80`)

Comment (drets) [inline src/Network/MQTT/Packet.hs]: 👍

File: src/Network/MQTT/Packet.hs [hunk 2]
```diff
@@ -36,22 +36,22 @@ data QoS
   | QoS2
   deriving (Eq, Show)
 
-newtype ClientIdentifier = ClientIdentifier Text
+newtype ClientIdentifier = ClientIdentifier { unClientIdentifier :: Text }
   deriving (Eq, Show)
 
 newtype PacketIdentifier = PacketIdentifier Word16
```
Comment (rasendubi) [inline src/Network/MQTT/Packet.hs]: It's worth to add `unPacketIdentifier` here, too.

File: src/Network/MQTT/Packet.hs [hunk 3]
```diff
@@ -134,7 +134,7 @@ data PubcompPacket
 data SubscribePacket
   = SubscribePacket
     { subsribePacketIdentifier :: !PacketIdentifier,
```
Comment (rasendubi) [inline src/Network/MQTT/Packet.hs]: add missing `c`. It should be `subscribe`

Comment (drets) [inline src/Network/MQTT/Packet.hs]: :see_no_evil:

File: src/Network/MQTT/Packet.hs (hunk 2 above)
Comment (drets) [inline src/Network/MQTT/Packet.hs]: 👍

**Pull request merged**